tensorbay/opendataset/VGGFace2/loader.py

```diff
@@ -0,0 +1,211 @@
+                )
+                segment.append(data)
+    return dataset
+
+
+def _get_data(
+    root_path: str,
+    segment_name: str,
+    line: str,
+    *,
+    all_classifications_id_to_name: Dict[str, str],
+    all_classifications: Dict[str, Classification],
+    all_box2ds: Dict[str, LabeledBox2D],
+    all_keypoint2ds: Dict[str, LabeledKeypoints2D],
+) -> Data:
+    line = line.rstrip("\n")
+    category_id, filename = line.split("/", 1)
+    data = Data(os.path.join(root_path, segment_name, category_id, filename))
+    category_name = all_classifications_id_to_name[category_id]
+    data.label.classification = all_classifications[category_name]
+    data.label.box2d = [all_box2ds[line]]
+    data.label.keypoints2d = [all_keypoint2ds[line]]
+
+    return data
+
+
+def _get_classifications(
+    label_path: str,
+    attribute_names: Tuple[str, ...],
+) -> Tuple[Dict[str, Classification], Dict[str, str]]:
+    all_classifications = {}
+    all_classifications_id_to_name = {}
+    with open(os.path.join(label_path, "identity_meta.csv"), encoding="utf-8") as fp:
+        lines = islice(csv.reader(fp), 1, None)
+        for line in lines:
+            # The normal format of each line of the file is
+            # '<class_id>,"<class_name>",<sample_num>,<flag>,<gender>,\n'
+            # but now there is an error type
+            # '<class_id>,"<class_,name>",<sample_num>,<flag>,<gender>\n'
+            if line[-1] != "":
+                # join the splitted "class_ and name"
+                line[1] = "".join(islice(line, 1, 3))
+                # Starting from the 4th element, each element moves forward one step
+                line.pop(2)
+                line[4] = line[4].rstrip("\n")
+            category_id = line[0]
+            category_name = line[1].strip('"')
+            attributes = dict(
+                zip(
+                    attribute_names,
+                    (
+                        int(line[2]),
+                        bool(int(line[3])),
+                        line[4],
+                    ),
+                )
+            )
+            all_classifications_id_to_name[category_id] = category_name
+            all_classifications[category_name] = Classification(
+                category=category_name,
+                attributes=attributes,
+            )
+    return all_classifications, all_classifications_id_to_name
+
+
+def _get_box2ds(label_path: str) -> Dict[str, LabeledBox2D]:
+    all_box2d_attributes: DefaultDict[str, AttributeType] = defaultdict(dict)
+    box2d_files = glob(os.path.join(label_path, "attributes", "*.txt"))
+    # The normal format of each line of the file is
+    # n000002/0032_01.jpg	0
+    # ...
+    for file in box2d_files:
+        attribute_name = os.path.basename(file).rstrip(".txt").split("-", 1)[1]
+        with open(file, encoding="utf-8") as fp:
+            for line in fp:
+                name_id, attribute_value = line.rstrip("\n").split("\t", 1)
```

Comment:
seems the `name_id` is different in different functions, here the `name_id` has extension, and in other function does not have extension?